src/app/tasks/tasks.component.ts

```diff
@@ -55,6 +59,31 @@ export class TasksComponent implements OnInit {
     this.tasksService.addTask({ ...task, assignee }).subscribe((res) => {
       this.tasksService.getTasks();
     });
+    this.sendNotifications();
+  }
+
+  sendNotifications(assignee: any = '') {
+    const link = '/teams/view';
+    const notificationDoc = ({ user, userPlanetCode }) => ({
+      user,
+      'message': 'You were assigned a new task',
+      link,
+      'type': 'newTask',
+      'priority': 1,
+      'status': 'unread',
+      'time': this.couchService.datePlaceholder,
+      userPlanetCode
+    });
+
+    if (assignee !== '' && assignee.userDoc) {
+      return this.couchService.findAll('notifications', findDocuments({ link, type: 'newTask', status: 'unread' })).pipe(
+        switchMap(() => {
+          const newNotifications = this.tasks.filter(assignee)
```

Comment:
You should filter on the result of the notifications, not the current list of tasks.

Also, `Array.filter()` takes a function as its first parameter.  The function returns `true` or `false` for each element of the array, and the result will be all elements in the array which return `true`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

Reply:
Thank you for the feedback. Will make these changes.

Reply:
Hey Paul, I pushed another revision to see if I am on a better track to solving this issue. Again, thank you for the feedback and sorry for the time it has taken.